### PR TITLE
chore: Fix project labels

### PR DESCRIPTION
**Pull request by mariammartins**
Hello folks,

This PR contains:

- changes in environment and env_code project labels

- change the variable from vpc_type to vpc

- Adding the use of the string "nul" in the vpc variable when the projects does not have a network

### CODE REVIEW

File: 0-bootstrap/variables.tf [hunk 1]
```diff
@@ -88,6 +88,12 @@ variable "bucket_tfstate_kms_force_destroy" {
   default     = false
 }
 
+variable "vpc" {
+  description = "The type of VPC to attach the project to. Possible options are base or restricted."
+  type        = string
+  default     = "null"
```
Comment (apeabody) [inline 0-bootstrap/variables.tf:94]: Hi @mariammartins! Is there an advantage to use string `"null"` rather than `null`?

File: 1-org/envs/shared/variables.tf [hunk 2]
```diff
@@ -97,6 +97,12 @@ variable "log_export_storage_retention_policy" {
   default = null
 }
 
+variable "vpc" {
+  description = "The type of VPC to attach the project to. Possible options are base or restricted."
+  type        = string
+  default     = "null"
```
Comment (renato-rudnicki) [inline 1-org/envs/shared/variables.tf:103]: I'm not sure why you are using `default = "null"`. If you want the default value would be `null`, it shouldn't use quotation marks. If you want the value to be empty, you should use `default = ""`.

File: 4-projects/modules/single_project/variables.tf [hunk 3]
```diff
@@ -72,10 +72,10 @@ variable "environment" {
   type        = string
 }
 
-variable "vpc_type" {
+variable "vpc" {
   description = "The type of VPC to attach the project to. Possible options are base or restricted."
   type        = string
-  default     = null
+  default     = "null"
```
Comment (renato-rudnicki) [inline 4-projects/modules/single_project/variables.tf:78]: See my previous comment.

**Review by renato-rudnicki — COMMENTED** [review]
Only small changes. LGTM

File: 1-org/modules/network/variables.tf [hunk 4]
```diff
@@ -45,6 +45,13 @@ variable "env_code" {
   description = "A short form of the environment to prepare within the Google Cloud organization (ex. d)."
 }
 
+variable "vpc" {
+  description = "The type of VPC to attach the project to. Possible options are base or restricted."
```
Comment (romanini-ciandt) [inline 1-org/modules/network/variables.tf:49]: Since you are explicit saying that the possible values for this variable are just `base` or `restricted`, I would recommend it to add a [custom validation](https://developer.hashicorp.com/terraform/language/values/variables#custom-validation-rules) to enforces that.

In addition to that, related to this "null" string, if this is really the text we want to add for projects that doesn't have VPC, we need to list it as a possible/valid string input option for the variable too.

File: 0-bootstrap/variables.tf (hunk 1 above)
Comment (mariammartins) [reply 0-bootstrap/variables.tf:94]: Hi @apeabody, as we have two types of network "base" and "restricted", the proposal is to use the string "null" in all projects that do not have a network, this way, its easier to check if all projects are labeled consistently.
Using an empty string deviates from the label standard off putting the network type in the label of projects with vpc.

Comment (apeabody) [reply 0-bootstrap/variables.tf:94]: Thanks @mariammartins! Just a suggestion, I personally might find `"none"` to be less confusing than `"null"` for no network.  Also might be safer given the overloaded meaning of `null`.

File: 2-environments/modules/env_baseline/variables.tf [hunk 5]
```diff
@@ -39,6 +39,12 @@ variable "monitoring_workspace_users" {
   type        = string
 }
 
+variable "vpc" {
+  description = "The type of VPC to attach the project to. Possible options are base or restricted."
+  type        = string
+  default     = "null"
```
[the remark this answers is not in the document]
Comment (apeabody) [reply 2-environments/modules/env_baseline/variables.tf:45]: FYI: I suspect using an empty string `""` for a label might result in reconciliation drift.

File: 0-bootstrap/variables.tf (hunk 1 above)
Comment (mariammartins) [reply 0-bootstrap/variables.tf]: @apeabody great suggestion, it makes more sense to use the value of none, so as not to cause confusion in the context of its use, change made.

**Comment by apeabody** [comment]
re-triggered `HubAndSpoke` (`default` already passed)

File: 0-bootstrap/README.md [hunk 6]
```diff
@@ -314,6 +314,7 @@ Each step has instructions for this change.
 | org\_project\_creators | Additional list of members to have project creator role across the organization. Prefix of group: user: or serviceAccount: is required. | `list(string)` | `[]` | no |
 | parent\_folder | Optional - for an organization with existing projects or for development/validation. It will place all the example foundation resources under the provided folder instead of the root organization. The value is the numeric folder ID. The folder must already exist. | `string` | `""` | no |
 | project\_prefix | Name prefix to use for projects created. Should be the same in all steps. Max size is 3 characters. | `string` | `"prj"` | no |
+| vpc | The type of VPC to attach the project to. Possible options are base or restricted. | `string` | `"none"` | no |
```
Comment (apeabody) [inline 0-bootstrap/README.md:317]: nit: should include `none`?

```suggestion
| vpc | The type of VPC to attach the project to. Possible options are none, base, or restricted. | `string` | `"none"` | no |
```

File: 0-bootstrap/variables.tf [hunk 7]
```diff
@@ -88,6 +88,17 @@ variable "bucket_tfstate_kms_force_destroy" {
   default     = false
 }
 
+variable "vpc" {
+  description = "The type of VPC to attach the project to. Possible options are base or restricted."
```
Comment (apeabody) [inline 0-bootstrap/variables.tf:92]: nit

```suggestion
  description = "The type of VPC to attach the project to. Possible options are none, base, or restricted."
```

File: 0-bootstrap/variables.tf [hunk 8]
```diff
@@ -88,6 +88,17 @@ variable "bucket_tfstate_kms_force_destroy" {
...
+  type        = string
+  default     = "none"
+
+  validation {
+    condition     = var.vpc == "base" || var.vpc == "restricted" || var.vpc == "none"
```
Comment (apeabody) [inline 0-bootstrap/variables.tf:97]: nit: you might consider using [contains ](https://developer.hashicorp.com/terraform/language/functions/contains) - more concise if additional options are added in the future.

```suggestion
    condition     = contains(["none", "base", "restricted"], var.vpc)
```

File: 1-org/envs/shared/README.md [hunk 9]
```diff
@@ -26,6 +26,7 @@
 | scc\_notification\_filter | Filter used to create the Security Command Center Notification, you can see more details on how to create filters in https://cloud.google.com/security-command-center/docs/how-to-api-filter-notifications#create-filter | `string` | `"state = \"ACTIVE\""` | no |
 | scc\_notification\_name | Name of the Security Command Center Notification. It must be unique in the organization. Run `gcloud scc notifications describe <scc_notification_name> --organization=org_id` to check if it already exists. | `string` | n/a | yes |
 | tfc\_org\_name | Name of the TFC organization | `string` | `""` | no |
+| vpc | The type of VPC to attach the project to. Possible options are base or restricted. | `string` | `"none"` | no |
```
Comment (apeabody) [inline 1-org/envs/shared/README.md:29]: nit
```suggestion
| vpc | The type of VPC to attach the project to. Possible options are none, base, or restricted. | `string` | `"none"` | no |
```

File: 1-org/envs/shared/backend.tf [hunk 10]
```diff
@@ -16,7 +16,7 @@
 
 terraform {
   backend "gcs" {
-    bucket = "UPDATE_ME"
+    bucket = "bkt-prj-b-seed-tfstate-fd64"
```
Comment (apeabody) [inline 1-org/envs/shared/backend.tf:19]: Was this change intentional?

File: 1-org/envs/shared/variables.tf [hunk 11]
```diff
@@ -97,6 +97,17 @@ variable "log_export_storage_retention_policy" {
   default = null
 }
 
+variable "vpc" {
+  description = "The type of VPC to attach the project to. Possible options are base or restricted."
```
Comment (apeabody) [inline 1-org/envs/shared/variables.tf:101]: nit
```suggestion
  description = "The type of VPC to attach the project to. Possible options are none, base, or restricted."
```

File: 1-org/envs/shared/variables.tf [hunk 12]
```diff
@@ -97,6 +97,17 @@ variable "log_export_storage_retention_policy" {
...
+  type        = string
+  default     = "none"
+
+  validation {
+    condition     = var.vpc == "base" || var.vpc == "restricted" || var.vpc == "none"
```
Comment (apeabody) [inline 1-org/envs/shared/variables.tf:106]: suggestion
```suggestion
    condition     = contains(["none", "base", "restricted"], var.vpc)
```

File: 1-org/modules/network/variables.tf [hunk 13]
```diff
@@ -45,6 +45,18 @@ variable "env_code" {
...
```
Comment (apeabody) [inline 1-org/modules/network/variables.tf:49]:
```suggestion
  description = "The type of VPC to attach the project to. Possible options are none, base, or restricted."
```

File: 1-org/modules/network/variables.tf [hunk 14]
```diff
@@ -45,6 +45,18 @@ variable "env_code" {
...
+  type        = string
+  default     = "none"
+
+  validation {
+    condition     = var.vpc == "base" || var.vpc == "restricted" || var.vpc == "none"
```
Comment (apeabody) [inline 1-org/modules/network/variables.tf:54]: suggestion
```suggestion
    condition     = contains(["none", "base", "restricted"], var.vpc)
```

File: 2-environments/modules/env_baseline/README.md [hunk 15]
```diff
@@ -10,6 +10,7 @@
...
 | remote\_state\_bucket | Backend bucket to load Terraform Remote State Data from previous steps. | `string` | n/a | yes |
 | tfc\_org\_name | Name of the TFC organization | `string` | n/a | yes |
+| vpc | The type of VPC to attach the project to. Possible options are base or restricted. | `string` | `"none"` | no |
```
Comment (apeabody) [inline 2-environments/modules/env_baseline/README.md:13]: nit
```suggestion
| vpc | The type of VPC to attach the project to. Possible options are none, base, or restricted. | `string` | `"none"` | no |
```

File: 2-environments/modules/env_baseline/variables.tf [hunk 16]
```diff
@@ -39,6 +39,17 @@ variable "monitoring_workspace_users" {
   type        = string
 }
 
+variable "vpc" {
+  description = "The type of VPC to attach the project to. Possible options are base or restricted."
+  type        = string
+  default     = "none"
+
+  validation {
+    condition     = var.vpc == "base" || var.vpc == "restricted" || var.vpc == "none"
```
Comment (apeabody) [inline 2-environments/modules/env_baseline/variables.tf:48]: suggestion
```suggestion
    condition     = contains(["none", "base", "restricted"], var.vpc)
```

File: 2-environments/modules/env_baseline/variables.tf [hunk 17]
```diff
@@ -39,6 +39,17 @@ variable "monitoring_workspace_users" {
   type        = string
 }
 
+variable "vpc" {
+  description = "The type of VPC to attach the project to. Possible options are base or restricted."
```
Comment (apeabody) [inline 2-environments/modules/env_baseline/variables.tf:43]:
```suggestion
  description = "The type of VPC to attach the project to. Possible options are none, base, or restricted."
```

File: 4-projects/modules/single_project/README.md [hunk 18]
```diff
@@ -21,10 +21,10 @@
 | secondary\_contact | The secondary email contact for the project | `string` | `""` | no |
 | shared\_vpc\_host\_project\_id | Shared VPC host project ID | `string` | `""` | no |
 | shared\_vpc\_subnets | List of the shared vpc subnets self links. | `list(string)` | `[]` | no |
+| vpc | The type of VPC to attach the project to. Possible options are base or restricted. | `string` | `"none"` | no |
```
Comment (apeabody) [inline 4-projects/modules/single_project/README.md:24]:
```suggestion
| vpc | The type of VPC to attach the project to. Possible options are none, base, or restricted. | `string` | `"none"` | no |
```

File: 4-projects/modules/single_project/variables.tf [hunk 19]
```diff
@@ -72,10 +72,15 @@ variable "environment" {
...
 
-variable "vpc_type" {
+variable "vpc" {
   description = "The type of VPC to attach the project to. Possible options are base or restricted."
   type        = string
-  default     = null
+  default     = "none"
+
+  validation {
+    condition     = var.vpc == "base" || var.vpc == "restricted" || var.vpc == "none"
```
Comment (apeabody) [inline 4-projects/modules/single_project/variables.tf:81]: suggestion
```suggestion
    condition     = contains(["none", "base", "restricted"], var.vpc)
```

**Review by apeabody — COMMENTED** [review]
Hi @mariammartins - A few minor suggestions and nits :)

File: 1-org/envs/shared/backend.tf (hunk 10 above)
Comment (mariammartins) [reply 1-org/envs/shared/backend.tf:19]: no!! I included it in the commit by mistake, sorry

File: 0-bootstrap/variables.tf (hunk 7 above)
Comment (mariammartins) [reply 0-bootstrap/variables.tf:92]: as "none" would be indicative of no network, and the only possible values ​​are "base" and "restricted", do you think that if we include none it would be necessary to put an indicative in the description, for example, "none" is used to indicate that there is no network created, or would it not be necessary?

Comment (apeabody) [reply 0-bootstrap/variables.tf:92]: @mariammartins - Just a suggestion. :) From the description only (without looking at the variable default or condition) a user might assume `vpc` MUST be one of `base` or `restricted`.  But likely a low risk.

File: 0-bootstrap/README-Jenkins.md
Comment (romanini-ciandt) [inline 0-bootstrap/README-Jenkins.md:1]: @mariammartins why did you delete this file? It was intentional?

Comment (mariammartins) [reply 0-bootstrap/README-Jenkins.md:1]: Definitely not, I must have made some confusion when resolving the conflicts, i will add it again, sorry and thanks!!

File: 0-bootstrap/variables.tf [hunk 20]
```diff
@@ -88,6 +88,17 @@ variable "bucket_tfstate_kms_force_destroy" {
...
 }
 
+variable "vpc" {
+  description = "The type of VPC to attach the project to. Possible options are none, base, or restricted."
+  type        = string
+  default     = "none"
+
+  validation {
+    condition     = contains(["none", "base", "restricted"], var.vpc)
+    error_message = "For vpc only `base`, `restricted`, and `none` are valid."
```
Comment (daniel-cit) [inline 0-bootstrap/variables.tf:98]:
```suggestion
    error_message = "For vpc, only `base`, `restricted`, and `none` are valid."
```

File: 1-org/envs/shared/variables.tf [hunk 21]
```diff
@@ -97,6 +97,17 @@ variable "log_export_storage_retention_policy" {
...
 }
 
+variable "vpc" {
+  description = "The type of VPC to attach the project to. Possible options are none, base, or restricted."
+  type        = string
+  default     = "none"
+
+  validation {
+    condition     = contains(["none", "base", "restricted"], var.vpc)
+    error_message = "For vpc only `base`, `restricted`, and `none` are valid."
```
Comment (daniel-cit) [inline 1-org/envs/shared/variables.tf:107]:
```suggestion
    error_message = "For vpc, only  `none`, `base`, or `restricted` are valid."
```

File: 1-org/modules/network/variables.tf [hunk 22]
```diff
@@ -45,6 +45,18 @@ variable "env_code" {
...
 }
 
+variable "vpc" {
+  description = "The type of VPC to attach the project to. Possible options are none, base, or restricted."
+  type        = string
+  default     = "none"
+
+  validation {
+    condition     = contains(["none", "base", "restricted"], var.vpc)
+    error_message = "For vpc only `base`, `restricted`, and `none` are valid."
```
Comment (daniel-cit) [inline 1-org/modules/network/variables.tf:55]:
```suggestion
    error_message = "For vpc, only  `none`, `base`, or `restricted` are valid."
```

File: 2-environments/modules/env_baseline/variables.tf [hunk 23]
```diff
@@ -39,6 +39,17 @@ variable "monitoring_workspace_users" {
...
 }
 
+variable "vpc" {
+  description = "The type of VPC to attach the project to. Possible options are none, base, or restricted."
+  type        = string
+  default     = "none"
+
+  validation {
+    condition     = contains(["none", "base", "restricted"], var.vpc)
+    error_message = "For vpc only `base`, `restricted`, and `none` are valid."
```
Comment (daniel-cit) [inline 2-environments/modules/env_baseline/variables.tf:49]:
```suggestion
    error_message = "For vpc, only  `none`, `base`, or `restricted` are valid."
```

File: 4-projects/modules/single_project/variables.tf [hunk 24]
```diff
@@ -72,10 +72,15 @@ variable "environment" {
...
-  description = "The type of VPC to attach the project to. Possible options are base or restricted."
+variable "vpc" {
+  description = "The type of VPC to attach the project to. Possible options are none, base, or restricted."
   type        = string
-  default     = null
+  default     = "none"
+
+  validation {
+    condition     = contains(["none", "base", "restricted"], var.vpc)
+    error_message = "For vpc only `base`, `restricted`, and `none` are valid."
```
Comment (daniel-cit) [inline 4-projects/modules/single_project/variables.tf:82]:
```suggestion
    error_message = "For vpc, only  `none`, `base`, or `restricted` are valid."
```

File: docs/TROUBLESHOOTING.md [hunk 25]
```diff
@@ -272,7 +272,7 @@ You will need to mark some Terraform resources as **tainted** in order to trigge
 
 1. In a terminal, navigate to the path where the error is being reported.
 
-   For example, if the unknown project ID is `prj-bu1-p-sample-base-abcd`, you should go to ./gcp-projects/business_unit_1/production (`business_unit_1` due to `bu1` and `production` due to `p`, see the Security Foundations [naming conventions](https://cloud.google.com/architecture/security-foundations/using-example-terraform#naming_conventions) for more information on the projects naming guideline).
+   For example, if the unknown project ID is `prj-bu1-p-sample-base-shared`, you should go to ./gcp-projects/business_unit_1/production (`business_unit_1` due to `bu1` and `production` due to `p`, see the Security Foundations [naming conventions](https://cloud.google.com/architecture/security-foundations/using-example-terraform#naming_conventions) for more information on the projects naming guideline).
```
Comment (eeaton) [inline docs/TROUBLESHOOTING.md:275]: For brand and url updates, please update text as below:

> (`business_unit_1` due to `bu1` and `production` due to `p`, see the Security Foundations [naming conventions](https://cloud.google.com/architecture/security-foundations/using-example-terraform#naming_conventions) for more information on the projects naming guideline).

change text and URL like:
"""
(`business_unit_1` due to `bu1` and `production` due to `p`, see [naming conventions](https://cloud.google.com/architecture/security-foundations/summary#naming-conventions) for more information.
"""

File: README.md [hunk 26]
```diff
@@ -194,38 +194,44 @@ Running this code as-is should generate a structure as shown below:
...
+        ├── prj-bu1-d-sample-base
+        ├── prj-bu1-d-sample-restrict
+        ├── prj-bu1-d-sample-peering
+    └── fldr-development-bu2
+        ├── prj-bu2-d-env-kms
+        ├── prj-bu2-d-sample-floating
+        ├── prj-bu2-d-sample-base
+        ├── prj-bu2-d-sample-restrict
+        └── prj-bu2-d-sample-peering
 └── fldr-non-production
```
Comment (eeaton) [inline README.md:209]: For all instances of "non-production" in resource ids, remove the hyphen like "nonproduction".

Context: Other environment names don't use a hyphen, using a hyphen for non-production makes the number of hyphens across environments inconsistent, so it's difficult to script over environments using regexp etc.

**Review by eeaton — APPROVED** [review]

**Comment by obriensystems** [comment]
- 2 symlinks modified in error during the non-production to nonproduction rename/refactor - the change is to ../common - which itself is a symlink so the double symlink works.  The other 6 of the 2x4 symlinks are unmodified as ../../
see change in #1107
https://github.com/terraform-google-modules/terraform-example-foundation/pull/1107/files#diff-bcfdfb477dea3f75d9d515aa61ce4f3836b223f11d6e4234122778963fbfef4a

https://github.com/terraform-google-modules/terraform-example-foundation/blob/master/3-networks-hub-and-spoke/envs/nonproduction/access_context.auto.tfvars
```
-../../access_context.auto.tfvars
+../shared/access_context.auto.tfvars
```
there is a 2nd symlink regression in the same directory in 
https://github.com/terraform-google-modules/terraform-example-foundation/blob/master/3-networks-hub-and-spoke/envs/nonproduction/common.auto.tfvars
```
-../../common.auto.tfvars
+../shared/common.auto.tfvars
```
I'll raise an issue shortly with the fix after I test a full upstream sync of the TEF.

**Comment by fmichaelobrien** [comment]
See https://github.com/terraform-google-modules/terraform-example-foundation/issues/1229
